### PR TITLE
Create attendee case when mobile worker created

### PR DESCRIPTION
**Pull request by SmittieC**
## Technical Summary

This PR includes the changes that I mentioned I am busy with in [this PR](https://github.com/dimagi/commcare-hq/pull/32748#issue-1625620273). Please see its description. My plan is to merge this PR into that one, and then that one to master. 

## Feature Flag

Attendance Tracking

### Safety story

- Local testing
- Unittests

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/events/tests/test_tasks.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+        # Now call the task again, which should not create new cases
+        sync_mobile_worker_attendees(domain_name=self.domain, user_id=self.webuser.user_id)
+
+        # Check total case count again
+        cases = AttendeeCase.objects.by_domain(self.domain, include_closed=True)
+        case_count = len(cases)
+        self.assertEqual(case_count, case_count_before)
+
+    def test_closed_cases_reopened_for_mobile_workers(self):
```
Comment (SmittieC) [inline corehq/apps/events/tests/test_tasks.py:90]: Please note that I have added a test to check the reopening of attendee cases when the `sync_mobile_worker_attendees` tasl is called, but there's not functionality to close the cases (i.e cases will not be closed when disabling the mobile worker attendees). I think we can do this in a separate PR?

File: corehq/apps/events/models.py [hunk 2]
```diff
@@ -55,14 +55,19 @@ class AttendanceTrackingConfig(models.Model):
...
     def toggle_mobile_worker_attendees(domain, value):
         config, _created = AttendanceTrackingConfig.objects.get_or_create(domain=domain)
         config.mobile_worker_attendees = value
         config.save()
 
+    @staticmethod
     def mobile_workers_can_be_attendees(domain):
-        config, _created = AttendanceTrackingConfig.objects.get_or_create(domain=domain)
-        return config.mobile_worker_attendees
+        try:
```
Comment (SmittieC) [inline corehq/apps/events/models.py:66]: I made this change because the way it previously was would have created an `AttendanceTrackingConfig` record simply by calling `mobile_workers_can_be_attendees`. Since we call this when creating a mobile worker - which can happen on a plan not supporting Attendance Tracking - this would have created a non-privilged record of `AttendanceTrackingConfig`

File: corehq/apps/events/tests/test_tasks.py (hunk 1 above)
Comment (zandre-eng) [reply corehq/apps/events/tests/test_tasks.py:90]: A different PR sounds good. It will be good to have this functionality as I can imagine users getting confused when they click the disable button but still see a list of mobile workers in their attendee list.

File: corehq/apps/users/views/mobile/users.py [hunk 3]
```diff
@@ -772,6 +778,28 @@ def create_mobile_worker(self, in_data):
...
+                new_case_block = get_case_block_for_user(
+                    mobile_worker_couch_user,
+                    created_by_user_id,
+                    attendee_case_type
+                )
+                return submit_case_blocks(
+                    [new_case_block.as_text()],
+                    domain=self.domain,
+                    user_id=created_by_user_id,
+                    device_id='UserCreated',
```
Comment (SmittieC) [inline corehq/apps/users/views/mobile/users.py:800]: Any better suggestions than `UserCreated`?

**Review by kaapstorm — APPROVED** [review]
Nice
